### PR TITLE
chore(actor) Remove more references to Actor

### PR DESCRIPTION
**Pull request by markstory**
This will help make removing writes to `Team.actor_id` and generating Actors lazily for users safer.

Refs HC-1176

### CODE REVIEW

File: src/sentry/api/fields/actor.py [hunk 1]
```diff
@@ -4,21 +4,18 @@
...
 from rest_framework import serializers
 
-from sentry.utils.actor import parse_and_validate_actor
+from sentry.utils.actor import ActorTuple, parse_and_validate_actor
 
 
 @extend_schema_field(field=OpenApiTypes.STR)
 class ActorField(serializers.Field):
     def __init__(self, *args, **kwds):
-        self.as_actor = kwds.pop("as_actor", False)
```
Comment (markstory) [inline src/sentry/api/fields/actor.py:13]: There is no usage of this anymore :tada:

File: src/sentry/api/helpers/group_index/update.py [hunk 2]
```diff
@@ -840,13 +841,13 @@ def handle_is_public(
 
 
 def handle_assigned_to(
-    assigned_actor: str,
-    assigned_by: str,
-    integration: str,
+    assigned_actor: ActorTuple,
+    assigned_by: str | None,
+    integration: str | None,
```
Comment (markstory) [inline src/sentry/api/helpers/group_index/update.py:846]: These types were wrong as we have tests passing in `None` and `ActorTuple` but never `str`.

File: src/sentry/integrations/slack/views/link_team.py [hunk 3]
```diff
@@ -167,7 +167,7 @@ def handle(self, request: Request, signed_params: str) -> HttpResponseBase:
         analytics.record(
             "integrations.identity_linked",
             provider="slack",
-            actor_id=team.actor_id,
+            actor_id=team.id,
```
Comment (markstory) [inline src/sentry/integrations/slack/views/link_team.py:170]: This is a behavior change but we won't have `actor_id` for long and `team.id` is the next best thing. The `actor_type` attribute should disambiguate what kind of id is being passed.

File: src/sentry/models/team.py [hunk 4]
```diff
@@ -250,16 +249,6 @@ def get_projects(self):
 
         return Project.objects.get_for_team_ids([self.id])
 
-    def get_member_actor_ids(self):
```
Comment (markstory) [inline src/sentry/models/team.py:253]: Unused untested method.

File: src/sentry/api/helpers/group_index/update.py [hunk 5]
```diff
@@ -840,13 +841,13 @@ def handle_is_public(
...
     group_list: list[Group],
     project_lookup: dict[int, Project],
     acting_user: User | None,
-) -> Actor | None:
+) -> Any | None:
```
Comment (wedamija) [inline src/sentry/api/helpers/group_index/update.py:850]: Maybe this should be `ActorSerializerResponse | None`?

**Review by wedamija — APPROVED** [review]

**Pull request merged**